Assets/MRTK/Providers/WindowsMixedReality/XR2018/Controllers/HPMotionController.cs

```diff
@@ -0,0 +1,271 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using Unity.Profiling;
+using UnityEngine;
+using UnityEngine.XR;
+using Microsoft.MixedReality.Input;
+using System;
+
+namespace Microsoft.MixedReality.Toolkit.WindowsMixedReality.Input
+{
+    public class MotionControllerState
+    {
+        public MotionControllerState(MotionController mc)
+        {
+            this.MotionController = mc;
+        }
+        public void Update(DateTime currentTime)
+        {
+            this.CurrentReading = MotionController.TryGetReadingAtTime(currentTime);
+        }
+        public MotionController MotionController { get; private set; }
+        public MotionControllerReading CurrentReading { get; private set; }
+
+    }
+
+    [MixedRealityController(
+        SupportedControllerType.HPMotionController,
+        new[] { Toolkit.Utilities.Handedness.Left, Toolkit.Utilities.Handedness.Right },
+        flags: MixedRealityControllerConfigurationFlags.UseCustomInteractionMappings)]
+    public class HPMotionController : WindowsMixedRealityController
+    {
+        public HPMotionController(TrackingState trackingState, Toolkit.Utilities.Handedness controllerHandedness, IMixedRealityInputSource inputSource = null, MixedRealityInteractionMapping[] interactions = null)
+            : base(trackingState, controllerHandedness, inputSource, interactions)
+        {
+        }
+
+        private static readonly ProfilerMarker UpdateControllerPerfMarker = new ProfilerMarker("[MRTK] HPController.UpdateController");
+
+        /// <summary>
+        /// Update the controller data from .
+        /// </summary>
+        public virtual void UpdateController(MotionControllerState controllerState)
```

Comment:
Another `internal` / `protected internal` opportunity, since we don't necessarily want something updating this other than the data provider.